library/src-bootstrapped/scala/IArray.scala

```diff
@@ -38,6 +38,166 @@ object opaques
     def (arr: IArray[Double]) length: Int = arr.asInstanceOf[Array[Double]].length
     def (arr: IArray[Object]) length: Int = arr.asInstanceOf[Array[Object]].length
     def [T](arr: IArray[T]) length: Int = arr.asInstanceOf[Array[T]].length
+
+    /** All the methods on Array[T] that don't mutate in-place can be used with IArray[T].
+     */
+    def [T, U >: T: ClassTag](arr: IArray[T]) ++(that: IArray[U]): IArray[U] =
+      (genericArrayOps(arr) ++ that.asInstanceOf[Array[U]]).asInstanceOf[IArray[U]]
```

Comment:
Are all the `asInstanceOf` still required?

Reply:
The `that.asInstanceOf[...]`, yes. The ones at the end are also required, but perhaps I can remove the brackets and it stills works.